backend/tracim_backend/models/data.py

```diff
@@ -108,6 +108,53 @@ class Workspace(DeclarativeBase):
     default_user_role = Column(
         Enum(WorkspaceRoles), nullable=False, server_default=WorkspaceRoles.READER.name,
     )
+    parent_id = Column(Integer, ForeignKey("workspaces.workspace_id"), nullable=True, default=None)
+    children = relationship(
+        "Workspace",
+        backref=backref("parent", remote_side=[workspace_id], order_by="Workspace.workspace_id",),
+    )
+
+    @property
+    def recursive_children(self) -> List["Workspace"]:
+        """typing.Listtyping.List
+        :return: list of children Workspace
+        :rtype Content
+        """
+        statement = text(
+            """
+            with RECURSIVE children_id as (
+                select workspaces.workspace_id as id from workspaces
+                where workspaces.parent_id = :workspace_id
+                union all
+                select workspaces.workspace_id as id from workspaces
+                join children_id c on c.id = workspaces.parent_id
+            )
+            select children_id.id as workspace_id from children_id;
```

Comment:
It wasn't possible to write this with SQLAlchemy?

Reply:
Complicated stuff to do with ORM, this is done in a similar way we do recursive query for content.
In fact this piece of code is no longer useful for this PR as we finally decide to not do recursive thing here, but we will later for https://github.com/tracim/tracim/issues/3549.
And this implementation is fully tested and not so trivial, so i do prefer to keep it here.

Reply:
Just to complete a bit about this point, the probable way to do this properly is by using sql expression language of Sqlalchemy : https://docs.sqlalchemy.org/en/13/core/tutorial.html .
The advantage is making the code a bit easier to maintain, The drawback of changing this code, is that it may took a lot of time to just rewrite something that work perfectly.

Reply:
@inkhey insert a TODO in the code. It's not required to refactor now.

Reply:
done. related issue here : https://github.com/tracim/tracim/issues/3670